File: KickbaseCore/Sources/KickbaseCore/Inspectables/InspectableSheet.swift
import SwiftUI

// A small wrapper that forwards to the real .sheet but provides a named modifier
// that can be extended in tests to conform to ViewInspector's PopupPresenter.

extension View {
    func sheet2<Sheet>(
        isPresented: Binding<Bool>, onDismiss: (() -> Void)? = nil,
        @ViewBuilder content: @escaping () -> Sheet
    ) -> some View where Sheet: View {
        return self.modifier(
            InspectableSheet(presented: isPresented, onDismiss: onDismiss, builder: content))
    }
}

struct InspectableSheet<Sheet>: SwiftUI.ViewModifier where Sheet: View {
    // Use different stored names so we can explicitly expose the protocol properties
    let presented: Binding<Bool>
    let onDismiss: (() -> Void)?
    let builder: () -> Sheet

    func body(content: Self.Content) -> some View {
        content.sheet(isPresented: presented, onDismiss: onDismiss, content: builder)
    }
}

#if canImport(ViewInspector)
    import ViewInspector
    extension InspectableSheet: PopupPresenter {
        public typealias Popup = Sheet
        public var isPresented: Binding<Bool> { presented }
        public var popupBuilder: () -> Sheet { builder }
    }
#endif

File: snapshots/skipstone/kickbasecore/KickbaseCore/destination/skipstone/KickbaseCore/src/main/kotlin/kickbase/core/InspectableSheet.kt
package kickbase.core

import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.saveable.Saver
import androidx.compose.runtime.saveable.rememberSaveable
import androidx.compose.runtime.setValue
import skip.lib.*

import skip.ui.*
import skip.foundation.*
import skip.model.*

fun <Sheet> View.sheet2(isPresented: Binding<Boolean>, onDismiss: (() -> Unit)? = null, content: () -> Sheet): View where Sheet: View = this.modifier(InspectableSheet(presented = isPresented, onDismiss = onDismiss, builder = content))

internal class InspectableSheet<Sheet>: skip.ui.ViewModifier where Sheet: View {
    // Use different stored names so we can explicitly expose the protocol properties
    internal val presented: Binding<Boolean>
    internal val onDismiss: (() -> Unit)?
    internal val builder: () -> Sheet

    override fun body(content: View): View {
        return ComposeBuilder { composectx: ComposeContext -> content.sheet(isPresented = presented, onDismiss = onDismiss, content = builder).Compose(composectx) }
    }

    constructor(presented: Binding<Boolean>, onDismiss: (() -> Unit)? = null, builder: () -> Sheet) {
        this.presented = presented.sref()
        this.onDismiss = onDismiss
        this.builder = builder
    }
}

